platform/ext/target/nordic_nrf/common/native_drivers/spu.c

```diff
@@ -151,3 +151,47 @@ void spu_regions_flash_config_non_secure_callable(uint32_t start_addr,
 		FLASH_NSC_REGION_FROM_ADDR(start_addr),
 		1 /* Lock */);
 }
+
+void spu_peripheral_config_secure(uint32_t periph_base_addr, bool periph_lock)
+{
+	/* Determine peripheral ID */
+	const uint8_t periph_id = NRFX_PERIPHERAL_ID_GET(periph_base_addr);
+
+	/* ASSERT checking that this is not an explicit Non-Secure peripheral */
+	NRFX_ASSERT((NRF_SPU->PERIPHID[periph_id].PERM &
+		SPU_PERIPHID_PERM_SECUREMAPPING_Msk) !=
+		(SPU_PERIPHID_PERM_SECUREMAPPING_NonSecure <<
+			SPU_PERIPHID_PERM_SECUREMAPPING_Pos));
+
+	nrf_spu_peripheral_set(NRF_SPU, periph_id,
+		1 /* Secure */,
+		1 /* Secure DMA */,
+		lock /* Lock */);
```

Comment:
Yeap, that's correct, this will help detecting trivial errors.
This is fixed now. Also a similar error in a function below.
I am also removing the inline comment - since the input is now parameterized.